python/Ganga/GPIDev/Lib/File/LocalFile.py

```diff
@@ -224,6 +224,19 @@ def remove(self):
 
         return
 
+    def get(self, localPath=''):
```

Comment:
I feel that we should keep the core of `IGangaFile` as simple as possible and that means both `get` and `put` doing one simple, predictable thing:

- `get()` copy all files which match `namePattern` from remote storage to `localDir`
- `put()` copy all files which match `namePattern` from `localDir` to remote storage

For `LocalFile` there's no difference between `locaDir` and remote storage so `get()` and `put()` should effectively be no-ops. If these functions always do this one predictable thing then we can depend on them and blindly call them on arbitrary `IGangaFile` instances (once we've ascertained whether it's an input or an output file).

If a user has a `DiracFile` which they want to be downloaded locally to some arbitrary location (e.g. `/home/person/my_analysis/data`) then they should set `DiracFile(..., localDir='/home/person/my_analysis/data')`.

Another confusion is that `IGangaFiles` do not represent files, they are instead serving a strange dual purpose as 1) declarative configuration (putting a `LocalFile` in `outputfiles` tells Ganga to run some different code path and do different things to if it were a `DiracFile`) and 2) a direction-undefined data pipe. One end of the pipe has some data (you must interpret the set schema attributes to work out which) and `get`/`put` push files through it. It can represent multiple files